### DISCUSSION TITLE
Feedback on using the SuperchainERC20 Starter Kit

### DISCUSSION THREAD

**Discussion by zainbacchus** [ethereum-optimism/ecosystem#586] [General]
Hi there! We noticed you recently forked the [SuperchainERC20 Starter Kit](https://console.optimism.io/build) and were curious if you had any feedback you could share.

We'd love to know:
- How'd you find the starter kit?
- What were you looking to do?
- Was it easy to follow?
- Did you deploy a SuperchainERC20 token? If so, where? If not, why?
- How can we improve it?

Thanks,
Zain & @pharger 

Requesting feedback from:
@bartolampb
@jeanevanss813
@nruthqxo2u
@ronaldperez7708056
@williamwalker2122094
@melinonoori
@xlinda2758
@edwardmiller8192501
@mylenegokmen
@reinhildtempl
@kevinedwards1335670
@paulmiller3600281
@hamzahaluk
@mishunir
@ReimagineTruth
@mdqst
@Cryptomoon7
@leopardracer
@yusufky63
@parseb
@lennardevertz
@diempi
@leonkanosekprouqe08
@garstkentxed5l1
@RamirezXxX
@shulsejigemi14
@varlandy7c5h8
@frolovqfm9
@lunzwetiwakejezo
@lilianamolfettadwgj3
@berhalterhofzm9p2
@heightgco9
@matthiesqopicukamalu
@roxanewzg0
@martiebuttersjwb
@lenzwillandrjp8
@fullertonkypercgc5gxj5a
@minmimuyemivayi
@aderholtdamolomev
@lenjonemcse
@HullyAI
@Wanmea
@Farzad3000
@erikside
@hulk0x
@BunsDev
@FractalVisionz
@FDDImax
@soldat13

**Comment by zainbacchus** [comment]
Requesting feedback from (cont):
@Ahverdil
@matizion
@robocups
@0xmjqarai
@naserkhomari
@adelalizadeh13
@naserya10
@Nassami1
@nsking7
@saeed21
@JRojeres
@khosroaraz
@bars26
@romz237
@whyte01
@hamedn53sv
@JaysonKit241990
@Josofia
@Bangbee13
@lardydellvunulobij
@silasteofila
@nataljalewis13354
@escamillamirad
@riikkaprez
@levensrabulas
@luigibolden
@balduinomalacan
@jonnadekok
@Phuong2554
@Hoangphuong2551
@bizzyisme146
@m4zeftiedwards
@nanceewilmott53
@kimbroshanon
@habcav856
@manupakhorshidul320
@rahcfbipomg
@gayetosijek
@dhdrgd97
@hoangphuong2550
@MrTrade59
@kgblackst0ne
@opthailand
@YINDEE9
@Agustinus13
@Chronique
@Sommatra
@jhont666
@Sanrababa
@Polonssk

**Comment by zainbacchus** [comment]
Requesting feedback from (cont):
@Morratt
@hamedsheibani
@Sfirnando
@olgamiran
@sakuragund
@pavel9213
@Nidish-Devops
@TheSalade
@YFiN99
@Ramees4
@mlinyekxy
@mikl10
@Guyincharge001
@MpcOS77
@tabshaikh
@Jensey35
@omahs
@tarunkhasnavis
@jessgusclark
@pariaskh
@s4rv4d

**Comment by TheSalade** [comment]
Hey,

- I found it by searching for the Superchain application on Google and the Superchain forum
- I am looking to build a Superchain Hyperlane Bridge
- Yeah, the template is really easy to use. I deployed the template token without difficulty
- I deployed a few to test my bridge. I started on the OP testnet
- For new features, I will certainly have more ideas in the future. I will come back with them

**Comment by Bangbee13** [comment]
Better i look the newly features right

**Comment by Agustinus13** [comment]
Yeah. That's a good plan. I agree with that.

Pada Kam, 12 Des 2024 03.33, TRTtheSalad ***@***.***> menulis:

**Comment by mlinyekxy** [comment]
what this?

Pada Kam, 12 Des 2024 02.56, Agustinus13 ***@***.***> menulis:

**Comment by parseb** [comment]
The tooling works. Feels good.
Seeing `totalSuppy` sync across chain feels magical and a bit scary.
The muti-super-chain deployment and environment feels maybe a bit too opinionated but it's probably just because it packs a lot. The downside being the environment is token focused so any higher level interactions are designed and tested elsewhere. Which is actually not that bad. Anyway, does what it was designed to do.

**Comment by Abuchtela** [comment]
1. How’d you find the starter kit?

I discovered the starter kit while researching tools for building on the Optimism Superchain. It was mentioned in a developer forum as a good starting point for creating scalable ERC20 tokens.

2. What were you looking to do?

My goal was to experiment with deploying an ERC20 token that could operate seamlessly across the Superchain ecosystem. I wanted to understand how interoperability works between chains and assess how the kit could be used for projects like public goods funding or Sybil-resistant tools.

3. Was it easy to follow?

The setup instructions were clear, and the code was well-organized. However, I think adding a detailed flow diagram showing how the token interacts across chains would help beginners grasp the multi-chain logic more intuitively.

4. Did you deploy a SuperchainERC20 token? If so, where? If not, why?

Yes, I deployed a SuperchainERC20 token on a testnet to experiment with the functionality. I used the Optimism Goerli testnet, which worked smoothly. However, for those new to deploying contracts, additional steps or troubleshooting tips (e.g., funding the wallet or handling deployment errors) might be helpful.

5. How can we improve it?
	•	Provide a step-by-step guide with visuals for non-technical users.
	•	Include common use cases or scenarios, such as integrating the token with dApps or bridging between chains.
	•	Add a section on testing and verifying the deployment. For instance, how to interact with the token using a block explorer or web3 tools.
	•	Offer more guidance on deploying on mainnet, including cost estimates and best practices.

**Comment by JaysonKit241990** [comment]
is there a step by step guide on how to do this? Deploying a
SuperchainERC20 ?

**Reply by Abuchtela** [reply]
Should be able to find the documentation here.... `https://docs.optimism.io/app-developers/starter-kit`

**Reply by Abuchtela** [reply]
To deploy a SuperchainERC20 token across the Superchain, you can utilize the SuperchainERC20 Starter Kit, which streamlines the creation, deployment, and testing of these tokens. Here’s a step-by-step guide:

1. Install Foundry

The SuperchainERC20 Starter Kit requires anvil, part of the Foundry toolchain. Follow the Foundry installation guide to set it up.

2. Clone the Repository

Clone the SuperchainERC20 Starter Kit repository and navigate into its directory:

git clone https://github.com/ethereum-optimism/superchainerc20-starter.git
cd superchainerc20-starter

3. Install Dependencies

Install the project dependencies using pnpm:

pnpm install

4. Initialize Environment Variables

Set up your environment variables by initializing the .env files:

pnpm init:env

5. Start the Development Environment

Launch the local development environment:

pnpm dev

6. Configure RPC URLs

Edit the .env files to include the RPC URLs of the chains where you intend to deploy your token. Ensure these URLs point to the correct network endpoints.

7. Modify Deployment Configuration

Adjust the deployment configuration parameters in the deploy-config file to suit your token’s specifications. This includes setting the token name, symbol, initial supply, and other relevant details.

8. Deploy the Token

With the configurations in place, deploy your SuperchainERC20 token:

pnpm deploy

This command will deploy the token contract to the specified chains, ensuring it has the same address across all deployments.

Best Practices:
	•	Use Create2 for Deployment: To maintain a consistent contract address across all chains, utilize the Create2 deployment method. This ensures your token remains fungible across different networks.
	•	Set Safe Permissions: Configure crosschainMint and crosschainBurn permissions carefully to prevent unauthorized minting or burning of tokens.

For more detailed information and advanced configurations, refer to the SuperchainERC20 Starter Kit documentation.

By following these steps, you can successfully deploy a SuperchainERC20 token across the Superchain, enabling seamless interoperability and consistent token behavior across multiple chains.

**Reply by Abuchtela** [reply]
> To deploy a SuperchainERC20 token across the Superchain, you can utilize the SuperchainERC20 Starter Kit, which streamlines the creation, deployment, and testing of these tokens. Here’s a step-by-step guide:
> 
> 1. Install Foundry
> 
> The SuperchainERC20 Starter Kit requires anvil, part of the Foundry toolchain. Follow the Foundry installation guide to set it up.
> 
> 2. Clone the Repository
> 
> Clone the SuperchainERC20 Starter Kit repository and navigate into its directory:
> 
> git clone https://github.com/ethereum-optimism/superchainerc20-starter.git
> cd superchainerc20-starter
> 
> 3. Install Dependencies
> 
> Install the project dependencies using pnpm:
> 
> pnpm install
> 
> 4. Initialize Environment Variables
> 
> Set up your environment variables by initializing the .env files:
> 
> pnpm init:env
> 
> 5. Start the Development Environment
> 
> Launch the local development environment:
> 
> pnpm dev
> 
> 6. Configure RPC URLs
> 
> Edit the .env files to include the RPC URLs of the chains where you intend to deploy your token. Ensure these URLs point to the correct network endpoints.
> 
> 7. Modify Deployment Configuration
> 
> Adjust the deployment configuration parameters in the deploy-config file to suit your token’s specifications. This includes setting the token name, symbol, initial supply, and other relevant details.
> 
> 8. Deploy the Token
> 
> With the configurations in place, deploy your SuperchainERC20 token:
> 
> pnpm deploy
> 
> This command will deploy the token contract to the specified chains, ensuring it has the same address across all deployments.
> 
> Best Practices:
> 	•	Use Create2 for Deployment: To maintain a consistent contract address across all chains, utilize the Create2 deployment method. This ensures your token remains fungible across different networks.
> 	•	Set Safe Permissions: Configure crosschainMint and crosschainBurn permissions carefully to prevent unauthorized minting or burning of tokens.
> 
> For more detailed information and advanced configurations, refer to the SuperchainERC20 Starter Kit documentation.
> 
> By following these steps, you can successfully deploy a SuperchainERC20 token across the Superchain, enabling seamless interoperability and consistent token behavior across multiple chains.

Info found here and much more....
`https://docs.optimism.io/app-developers/starter-kit `

@Abuchtela

**Comment by naserkhomari** [comment]
Yeah. That's a good plan. I agree with that

**Comment by Agustinus13** [comment]
Yes, I agree.

Pada Sel, 28 Jan 2025 06.59, naserkhomari ***@***.***>
menulis:

**Comment by Ramees4** [comment]
Good 👍

**Comment by robocups** [comment]
> Hi there! We noticed you recently forked the [SuperchainERC20 Starter Kit](https://console.optimism.io/build) and were curious if you had any feedback you could share.
> 
> We'd love to know:
> 
> * How'd you find the starter kit?
> * What were you looking to do?
> * Was it easy to follow?
> * Did you deploy a SuperchainERC20 token? If so, where? If not, why?
> * How can we improve it?
> 
> Thanks, Zain & @pharger
> 
> Requesting feedback from: @bartolampb @jeanevanss813 @nruthqxo2u @ronaldperez7708056 @williamwalker2122094 @melinonoori @xlinda2758 @edwardmiller8192501 @mylenegokmen @reinhildtempl @kevinedwards1335670 @paulmiller3600281 @hamzahaluk @mishunir @ReimagineTruth @mdqst @Cryptomoon7 @leopardracer @yusufky63 @parseb @lennardevertz @diempi @leonkanosekprouqe08 @garstkentxed5l1 @RamirezXxX @shulsejigemi14 @varlandy7c5h8 @frolovqfm9 @lunzwetiwakejezo @lilianamolfettadwgj3 @berhalterhofzm9p2 @heightgco9 @matthiesqopicukamalu @roxanewzg0 @martiebuttersjwb @lenzwillandrjp8 @fullertonkypercgc5gxj5a @minmimuyemivayi @aderholtdamolomev @lenjonemcse @HullyAI @Wanmea @Farzad3000 @erikside @hulk0x @BunsDev @FractalVisionz @FDDImax @soldat13

I think everything is fine, I just had some errors during deployment and compilation, which I later realized were due to my own libraries.

**Reply by Josofia** [reply]
I think everything is fine, I’ve not deployed a SuperchainERC20 token because I’m still gathering my funds for liquidity
